common/lib/xmodule/xmodule/modulestore/tests/test_publish.py

```diff
@@ -139,3 +152,614 @@ def test_publish_draft_delete(self):
+        """
+        for block_data in block_list:
+            block_params = self.course_db.get(block_data)
+            self.assertIsNotNone(block_params)
+            (block_type, block_id) = block_data
+            regex_func = None
+            if block_type == 'chapter':
+                regex_func = self._make_chapter_matching_regex
+            elif block_type == 'sequential':
+                regex_func = self._make_sequential_matching_regex
+                block_params.update({'seq_children_published': kwargs.get('seq_children_published')})
+            elif block_type == 'vertical':
+                regex_func = self._make_vertical_matching_regex
+            elif block_type == 'html':
+                regex_func = self._make_html_unit_matching_regex
+            block_regex = regex_func(
+                self.course_key,
+                draft=draft,
+                **block_params
+            )
+            self.assertOLXContent(block_type, block_id, draft=draft, xml_re=block_regex)
+            self.assertOLXMissing(block_type, block_id, draft=(not draft))
+
+    def assertOLXIsDraft(self, block_list, **kwargs):
+        """
+        Check that all blocks in the list are draft blocks in the OLX format when the course is exported.
+        """
+        self._assertOLXBase(block_list, draft=True, **kwargs)
+
+    def assertOLXIsPublished(self, block_list, **kwargs):
+        """
+        Check that all blocks in the list are published blocks in the OLX format when the course is exported.
+        """
+        self._assertOLXBase(block_list, draft=False, **kwargs)
+
+    def publish(self, block_type, block_id):
+        """
+        Get an item, publish it, and shift to a new course export dir.
+        """
+        # Get the specified test item from the draft branch.
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred):
+            test_item = self.store.get_item(self.course_key.make_usage_key(block_type=block_type, block_id=block_id))
+        # Publish the draft item to the published branch.
+        self.store.publish(test_item.location, self.user_id)
+        # Since the elemental operation is now complete, shift to the post-operation export directory name.
+        self.export_dir = self.EXPORTED_COURSE_AFTER_DIR_NAME
+        self.before_publish = False
+
+
+@ddt.ddt
+class ElementalPublishingTests(UniversalTestProcedure):
+    """
+    Tests for the publish() operation.
+    """
+    @ddt.data(DRAFT_MODULESTORE_SETUP, MongoModulestoreBuilder())
+    def test_publish_old_mongo_unit(self, modulestore_builder):
+        with self._setup_test(modulestore_builder, self.course_key) as (
+            self.root_export_dir, self.contentstore, self.store, self.course
+        ):
+            block_type = 'html'
+            block_id = 'unit00'
+            # In old Mongo, you can successfully publish an item whose parent
+            # isn't published.
+            self.publish(block_type, block_id)
```

Comment:
(And I think it would be helpful to have that assertion of structure/format be in the test, so that it clearly ties `publish` to the particular format of a published item)